src/main/java/Parser.java

```diff
@@ -0,0 +1,234 @@
+
+        StringBuilder sb = new StringBuilder();
+
+        //Loading stored task into current list
+        Scanner pastScan = new Scanner(new FileReader("/Users/teojunhong/JavaProject/2103T/duke/savedList.txt"));
+        loadExisting(pastScan);
+
+        //Scanner obj for input
+        Scanner sc = new Scanner(System.in);
+
+        //Loop till user input 'bye'
+        String input = "";
+        if (sc.hasNextLine()) {
+            input = sc.nextLine();
+        }
+        while (!input.toLowerCase().equals("bye")) {
+
+            //adding items to arraylist for listing purpose
+            if (input.toLowerCase().contains("done")) {
+                try {
+                    //mark task as done, change cross to tick
+                    int itemNum = Integer.parseInt(input.substring(input.length() - 1));
+                    Task curr = items.get(itemNum - 1);
+                    curr.markAsDone();
+                    //forming the message
+                    user.doneMessage(curr);
+                    sb.setLength(0);
+                    input = sc.nextLine();
+                } catch (IndexOutOfBoundsException e) {
+                    user.indexError();
+                    input = sc.nextLine();
+                } catch (NumberFormatException e) {
+                    user.emptyError();
+                    input = sc.nextLine();
+                }
+            } else if (input.toLowerCase().contains("delete")) {
+                try {
+                    //delete task
+                    int itemNum = Integer.parseInt(input.substring(input.length() - 1));
+                    deleteTask(itemNum);
+                    input = sc.nextLine();
+                } catch (IndexOutOfBoundsException e) {
+                    user.indexError();
+                    input = sc.nextLine();
+                } catch (NumberFormatException e) {
+                    user.emptyError();
+                    input = sc.nextLine();
+                }
+            } else if (input.toLowerCase().equals("list")) {
+                //listing all task in current list
+                user.listTask(items);
+                input = sc.nextLine();
+            } else if (input.toLowerCase().contains("find")) {
+                String searchTerm = input.substring(5);
+                user.findTask(items, searchTerm);
+                input = sc.nextLine();
+            } else {
+                //generate task and store into list
+                generateTask(input);
+                input = sc.nextLine();
+            }
+        }
+
+        //store current tasks into local file for future reference
+        storeCurrent(items);
+
+        //Concluding Message
+        user.conclusion();
+    }
+    */
+
+
+    public boolean checkDuplicate(Task curr) {
+        boolean result = false;
+        for (int i = 0; i < items.size(); i++) {
+            Task test = items.get(i);
+            if (test.getClass().equals(curr.getClass())) {
```

Comment:
is this unnecessary work? Perhaps checking of class has already been done in the `equals` method